### DISCUSSION TITLE
Naming around Keys (Possibly for v.2.0.0)

### DISCUSSION THREAD

**Discussion by keithrfung** [microsoft/electionguard#90] [Ideas]
Here's a thought for the naming of the keys that I'd like to have a discussion around. 

As of right now, we have stuck with this concept for a while of public and private keys copying typical encryption keys like RSA. This is pretty great EXCEPT when we hit the partial concepts.

Here's a thought I had to move towards more Plain English. 

Election Public Key => Encryption Key
Election Private Key => Decryption Key
Election Partial Key Backup => Partial Decryption Key
Joint Key => Joint Encryption Key

I think this greatly improves wording and concepts

### Before
1. Combine Guardian's Election Public Keys to create a Joint Key to encrypt ballots
2. Guardian sends election partial key backup to other guardians
3. Available guardians combine election partial key backup to form election private key of missing guardian

### After
1. Combine Guardian's Encryption Keys to create a Joint Encryption Key to encrypt ballots
2. Guardian sends partial decryption key to other guardians
3. Available guardians combine partial decryption key to form decryption key of missing guardian

**Assumptions**
- Removal of `AuxiliaryKeyPair`
- There is a way encrypting write in's 

### Before

```py
class ElectionKeyPair:
     public_key: PUBLIC_KEY
     private_key: PRIVATE_KEY
     
class ElectionPartialKeyBackup
      owner_id: str
      designated_id: str
      value: str
```

### After

```py
class GuardianKeyPair:
     encryption_key: PUBLIC_KEY
     decryption_key: PRIVATE_KEY
     
class PartialDecryptionKey
      owner_id: str
      designated_id: str
      value: str
```

### Before

**Key Ceremony**
Round 1: Guardian shares their election public key
Round 2: Guardian shares partial election key backup to other Guardians
Round 3: Guardian verifies their received partial election key backup works (edited) 

**Tally Ceremony**
Available Guardians decrypt their share of tally with their election private keys
Available Guardians each use their partial election key backup to decrypt a partial of the missing guardians share of the tally

### After

**Key Ceremony**
Round 1: Guardian shares their encryption key
Round 2: Guardian shares partial decryption key to other Guardians
Round 3: Guardian verifies their received partial decryption key works (edited) 

**Tally Ceremony**
Available Guardians decrypt their share of tally with their decryption keys
Available Guardians each use their partial decryption key to decrypt a partial of the missing guardians share of the tally

**Comment by keithrfung** [comment]
CC @benaloh and @rc-ms

**Reply by benaloh** [reply]
A couple of things ...

- We NEVER form a missing guardian's private key.  We instead form a missing guardian's partial decryption of data.  This is an important distinction.
- Do we like "Ballot Encryption Key" instead of "Joint Encryption Key"?

**Reply by keithrfung** [reply]
Updated
